common/src/main/java/com/wynntils/services/mapdata/providers/builtin/CharacterProvider.java

```diff
@@ -4,39 +4,86 @@
 import com.wynntils.models.character.event.CharacterMovedEvent;
+import com.wynntils.services.hades.HadesUser;
+import com.wynntils.services.hades.event.HadesEvent;
+import com.wynntils.services.hades.event.HadesUserEvent;
 import com.wynntils.services.mapdata.attributes.AbstractMapAttributes;
 import com.wynntils.services.mapdata.attributes.FixedMapVisibility;
 import com.wynntils.services.mapdata.attributes.type.MapAttributes;
+import com.wynntils.services.mapdata.attributes.type.MapDecoration;
+import com.wynntils.services.mapdata.attributes.type.MapIcon;
 import com.wynntils.services.mapdata.attributes.type.MapVisibility;
 import com.wynntils.services.mapdata.type.MapCategory;
 import com.wynntils.services.mapdata.type.MapFeature;
 import com.wynntils.services.mapdata.type.MapLocation;
+import com.wynntils.utils.colors.CustomColor;
 import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.mc.SkinUtils;
 import com.wynntils.utils.mc.type.Location;
+import com.wynntils.utils.render.Texture;
+import com.wynntils.utils.render.buffered.BufferedRenderUtils;
+import com.wynntils.utils.render.type.HealthTexture;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Locale;
 import java.util.Optional;
 import java.util.stream.Stream;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.resources.ResourceLocation;
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 public class CharacterProvider extends BuiltInProvider {
-    public static final CharacterLocation CHARACTER_LOCATION = new CharacterLocation();
-    private static final List<MapFeature> PROVIDED_FEATURES = List.of(CHARACTER_LOCATION);
+    private static final CharacterLocation CHARACTER_LOCATION = new CharacterLocation();
+    private static final List<HadesPlayerLocation> PLAYERS = new ArrayList<>();
+
+    // Provided features and categories
     private static final List<MapCategory> PROVIDED_CATEGORIES = List.of(new PlayersCategory());
 
     @SubscribeEvent
     public void onCharacterMove(CharacterMovedEvent e) {
         notifyCallbacks(CHARACTER_LOCATION);
     }
 
+    @SubscribeEvent
+    public void onHadesAuthenticatedEvent(HadesEvent.Authenticated event) {
+        reloadHadesUsers();
+    }
+
+    @SubscribeEvent
+    public void onHadesDisconnectedEvent(HadesEvent.Disconnected event) {
+        reloadHadesUsers();
+    }
+
+    @SubscribeEvent
+    public void onHadesUserAddedEvent(HadesUserEvent.Added event) {
+        reloadHadesUsers();
+    }
+
+    @SubscribeEvent
+    public void onHadesUserRemovedEvent(HadesUserEvent.Removed event) {
+        reloadHadesUsers();
+    }
+
+    private void reloadHadesUsers() {
+        PLAYERS.forEach(this::notifyCallbacks);
+        PLAYERS.clear();
+        Services.Hades.getHadesUsers().map(HadesPlayerLocation::new).forEach(PLAYERS::add);
+    }
+
     @Override
     public String getProviderId() {
-        return "character";
+        return "characters";
```

Comment:
Maybe we should rename this `PlayerProvider` and have `players` as provider id? We've tried to reserve "character" for the user playing the game, and call everyone else "player", so we should try and stick with that.

(I realize I named this wrong)

Reply:
Do you want to separate Hades into PlayerCategory, and keep the "character/us" here?

Reply:
Also, can you explain what the `CharacterLocation` provides? It has no texture, no label, and we still render the "arrow" on the map separately? Do you want to replace that special rendering with this provider, but did not finish writing it?

Reply:
> Do you want to replace that special rendering with this provider, but did not finish writing it?

That was my thinking, but in retrospect I'm not sure it was a good idea. I'd say at this point it is better to remove it, and focus on making this provider do the "other players" things. If we figure out a need for the user marker to be a map feature, we can restore it.

Reply:
Fixed.